src/ruby/lib/grpc/generic/rpc_server.rb

```diff
@@ -413,9 +428,9 @@ def implemented?(an_rpc)
 
     # handles calls to the server
     def loop_handle_server_calls
-      fail 'not running' unless @running
+      fail 'not started' if running_state == :not_started
       loop_tag = Object.new
-      until stopped?
+      while running_state == :running
```

Comment:
is there a concept of volatile in ruby? technically just reading  running_state is unsafe.